crates/biome_css_analyze/src/lint/nursery/no_descending_specificity.rs

```diff
@@ -0,0 +1,209 @@
+        }
+        _ => None,
+    }
+}
+
+/// This function traverses the CSS rules starting from the given rule and checks for selectors that have the same tail selector.
+/// For each selector, it compares its specificity with the previously encountered specificity of the same tail selector.
+/// If a lower specificity selector is found after a higher specificity selector with the same tail selector, it records this as a descending selector.
+fn find_descending_selector(
+    rule: &CssSemanticRule,
+    model: &SemanticModel,
+    visited_rules: &mut BTreeSet<RuleId>,
+    visited_selectors: &mut BTreeMap<String, (TextRange, Specificity)>,
+    descending_selectors: &mut Vec<DescendingSelector>,
+) {
+    if visited_rules.contains(&rule.id) {
+        return;
+    } else {
+        visited_rules.insert(rule.id);
+    };
+
+    for selector in &rule.selectors {
+        let tail_selector = if let Some(s) = find_tail_selector(&selector.original) {
+            s
+        } else {
+            continue;
+        };
+
+        if let Some((last_textrange, last_specificity)) = visited_selectors.get(&tail_selector) {
+            if last_specificity > &selector.specificity {
+                descending_selectors.push(DescendingSelector {
+                    high: (*last_textrange, last_specificity.clone()),
+                    low: (selector.range, selector.specificity.clone()),
+                });
+            }
+        } else {
+            visited_selectors.insert(
+                tail_selector,
+                (selector.range, selector.specificity.clone()),
+            );
+        }
+    }
+
+    for child_rule in rule
+        .child_ids
+        .iter()
+        .filter_map(|id| model.get_rule_by_id(*id))
+    {
+        find_descending_selector(
+            child_rule,
+            model,
+            visited_rules,
+            visited_selectors,
+            descending_selectors,
+        );
+    }
+}
+
+impl Rule for NoDescendingSpecificity {
+    type Query = Semantic<CssRoot>;
+    type State = DescendingSelector;
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let model = ctx.model();
+        let mut visited_rules = BTreeSet::new();
+        let mut visited_selectors = BTreeMap::new();
+        let mut descending_selectors = Vec::new();
+        for rule in model.rules() {
+            find_descending_selector(
+                rule,
+                model,
+                &mut visited_rules,
+                &mut visited_selectors,
+                &mut descending_selectors,
+            );
+        }
+
+        descending_selectors
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, node: &Self::State) -> Option<RuleDiagnostic> {
+        //
+        // Read our guidelines to write great diagnostics:
+        // https://docs.rs/biome_analyze/latest/biome_analyze/#what-a-rule-should-say-to-the-user
+        //
```

Comment:
Please remove this comment

Reply:
Fixed at https://github.com/biomejs/biome/pull/4097/commits/0dbdbd1a03d8d48caf086edc83a4b7916bc5e7fe

Reply:
Fixed at https://github.com/biomejs/biome/pull/4097/commits/0dbdbd1a03d8d48caf086edc83a4b7916bc5e7fe